### PR TITLE
#2329 Versioned program for `FakeMaven`

### PR DESCRIPTION
**Pull request by maxonfjvipon**
What's done:
- Added versioned program to `FakeMaven`
- Refactored tests that test versioning

---

## PR-Codex overview
This PR focuses on adding versioned programs to the EO Maven Plugin. 

### Detailed summary
- Added `withVersionedProgram` method to `FakeMaven` for versioned programs
- Updated test cases in `OptimizeMojoTest`, `VersionsMojoTest`, `DiscoverMojoTest`, and `ProbeMojoTest` to use versioned programs
- Updated program versions in test cases

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: eo-maven-plugin/src/test/java/org/eolang/maven/ProbeMojoTest.java [hunk 1]
```diff
@@ -69,6 +64,11 @@
 @SuppressWarnings("PMD.AvoidDuplicateLiterals")
 @ExtendWith(OnlineCondition.class)
 final class ProbeMojoTest {
+    /**
+     * Fake hashes.
+     */
+    private static final Map<String, CommitHash> HASHES = new CommitHashesMap.Fake();
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/test/java/org/eolang/maven/ProbeMojoTest.java:70]: @maxonfjvipon [personal opinion] It is potential concurrency problem. All our tests work in parallel, so they can fulfil `HASHES` simultaneously. As I understand `CommitHashesMap.Fake` is thread-safe, at least for now, but we can change it in the future, which can lead to some problems.
Moreover the benefit which you get from the `static` field is not significant.
___
Of course, you can argue, that in the future we can make `CommitHashesMap.Fake` thread-safe,  so I leave it up to you.

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo please have a look one more time

**Review by volodya-lombrozo — APPROVED** [review]
@maxonfjvipon Looks good to me, thank you!

**Comment by maxonfjvipon** [comment]
@yegor256 please have a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**